src/main/java/org/jboss/msc/service/ServiceControllerImpl.java

```diff
@@ -571,12 +570,12 @@ private void transition(final ArrayList<Runnable> tasks) {
                         dependenciesDemanded = false;
                     }
                     getListenerTasks(transition, tasks);
-                    tasks.add(new DependencyStoppedTask(getDependents()));
```

Comment:
The fact that we're no longer using a snapshot of the dependents array is a functional change I think, because now we're iterating dependents outside of the lock.